### ISSUE TITLE
Drop jquery from core bokehjs

### ISSUE THREAD

**Issue by mattpap** [bokeh/bokeh#5625]
This is another milestone in making (core) bokehjs more maintainable and smaller. jquery will remain a dependency of bokehjs' widgets (at least data table's, we will see about the other). Setting milestone to 0.12.5, but this will likely take more than that.

**Comment by bryevdv** [comment]
We should try to do a search on GH, SO, the mailing list, etc. to see how often `Bokeh.$` comes up. It needs to be noted as a migration in any case, but I'd like some idea of how disruptive removing that might be to existing code. I feel like we demonstrated its usage at least a few times in the past.

**Comment by mattpap** [comment]
`Bokeh.$` is already deprecated (since #5532), so anyone affected may get to us with complaints anyway. However, we got multiple complaints on the contrary that we shouldn't include jquery with bokeh. At first it was because bokehjs' jquery was getting in conflict with jquery included in a web page. That was resolved and another batch of complaints was that jquery is already included everywhere and web sites have to include another copy included in bokehjs (which never was too fit). Given that actually most web sites come with jquery, removal of `Bokeh.$` shouldn't be an issue.

Just to note, this issue isn't about dropping `Bokeh.$` yet, but making core bokehjs independent of jquery, i.e. use native APIs (DOM), TSX, etc.
